### PR TITLE
Advanced emote contrib

### PR DESCRIPTION
**Pull request by rhicora**
#### Brief overview of PR changes/additions

- Added the `advanced_emote` contrib, which allows regular players to write emotes 
  in actor-stance using the `I`, `My`, and `Mine` commands.
- Added `$prot()` inlinefunc for pronoun parsing in actor-stance strings using
  `msg_contents` with specific targets ("and nods to $prot(pronoun, target)").
- Added optional dependency nltk, only necessary if you use advanced_emote, and
  the convenience script get_nltk_data.py for downloading its dataset.

image
image

#### Motivation for adding to Evennia

Allows any player on an Evennia MU that uses this contrib to emote in Actor Mode. To use, `self.add(CmdI)` to your basic Character's command set (you'll need to `from evennia.contrib.advanced_emote.actor_emote import CmdI` to get that command).

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ build-backend = "setuptools.build_meta"
 
 [project]
 name = "evennia"
-version = "1.1.1"
+version = "1.2.0"
```
[the remark this answers is not in the document]
Comment (rhicora) [reply pyproject.toml:7]: Ok, good.

**Comment by rhicora** [comment]
> Please split this into two PRs though; they are unrelated features from one another.

$prot is intentionally separated; it can be accessed with or without usage of the contrib! If someone is working on something that requires $prot independently of this I'm happy to split this to a separate PR.

File: evennia/contrib/advanced_emote/actor_emote.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        args = self.args
+        if args and not args[0] in ["'", ","]:
+            args = " %s" % args.strip()
+        self.args = args
+
+    def func(self):
+        """Hook function"""
+        if not self.args:
+            """They were probably trying to get at their inventory."""
+            """Todo: make this cause the caller to call inv directly."""
```
[the remark this answers is not in the document]
Comment (rhicora) [reply evennia/contrib/advanced_emote/actor_emote.py:41]: Agree, will change how this decision works, probably with a setting that defaults to `i` for `inv`.

**Comment by Griatch** [comment]
(from discord, for completeness)
> Thanks for the review @Griatch! I was not sure the convention on versioning and whether we version bump directly from major feature branches or not. If you do that later on in the process then that's better practice, imo! I'll remove the version bump.

> $prot is a prereq for the other feature but it is released outside of the contrib (it's in evennia/utils/funcparser.py regardless of whether you import advanced emote) since it's a broadly useful inlinefunc that doesn't get in the way of anything else. I can split them into two PRs if that's really necessary, but I think they're separated enough already? Unless you really want $prot for some reason while we continue to develop advanced emote.

> Agree on leaving i behavior as is unless the user decides to change it. Would you be open to a setting that specifies whether to use I for inv or not? (edited)

--- 
To answer your questions:

- I feel `$prot` is something that can be used and evaluated (and merged) separately. While the contrib depends on it, the opposite is not true. It also doesn't require much extra review. So breaking this into a separate PR is the way to go imo.
- As it stands the contrib-part of the PR will require more work before it can be merged. While the idea is good, I'm not convinced about this implementation and would like to see more discussion of alternatives to having to add an additional `nltk` requirement (I also question the practice of adding code in the `scripts/` folder, way outside of the contrib). See [the guide for contribs](https://www.evennia.com/docs/latest/Contribs/Contribs-Guidelines.html) for how a contrib should be structured (e.g. a `README.md` is required). But we can leave that discussion to the separate PR :) 
- Having a new default setting just for using `i` is not really viable I think. The best way would be to simply inform the user (in docstring / README) that they need to remove the `i` alias for the inventory command.  The unfinished TODO in the contrib (for forwarding just plain 'i' to `inventory`) is an interesting way to get the best of both worlds.

**Comment by Griatch** [comment]
@rhicora Any updates on this? :slightly_smiling_face:

**Comment by Griatch** [comment]
Ping.

**Comment by Griatch** [comment]
@rhicora If we can split this into two, one of them should be ready to merge :)

**Comment by Griatch** [comment]
Closing this since there's no further updates for several months. If you feel you want to finish this one up, reply here and I'd be happy to open this up again.

**Pull request closed without merging**